fs/dafsclient/rmtclient.cpp

```diff
@@ -714,13 +711,36 @@ void CRemoteBase::connectSocket(SocketEndpoint &ep, unsigned connectTimeoutMs, u
             }
             else
                 socket.setown(ISocket::connect(ep));
-            if (ep.port == securitySettings.queryDaFileSrvSSLPort())
+            if (secure)
             {
 #ifdef _USE_OPENSSL
                 Owned<ISecureSocket> ssock;
                 try
                 {
-                    ssock.setown(createSecureSocket(socket.getClear(), nullptr));
+                    // instead of creating CRemoteFile/CRemoteBase with a secret, it could instead lookup, and create the ISecureSocketContext and pass that
+                    // into the ctor. That would avoid having to look the secret up, extract the certs and recreate the context on each connection.
+                    // The context would be 'owned' by the hook, and would expire when the mappings are removed (when removeMappedDafileSrvSecrets is called).
```

Comment:
NB: @ghalliday @mckellyln - in a subsequent PR I will be refactoring the below so that it uses the new ISecret interface and caching the security contexts